src/native/corehost/corehost.cpp

```diff
@@ -238,7 +255,7 @@ int exe_start(const int argc, const pal::char_t* argv[])
             // This check exists to provide an error message for UI apps when running 3.0 apps on 2.0 only hostfxr, which doesn't support error writer redirection. 
```

Comment:
Do we actually need to do this anymore now that we are not relying on a custom prefix to show a different message?

Reply:
Oh, I guess we are only using it for the URL now? Maybe the dialog should just default to `get_download_url()` + `apphost_version=...`  if it doesn't parse one out of the buffered error messages.

Reply:
It's actually also used to write something meaningful to eventlog. On Windows we always redirect error writer. If it's a console app, it's only used to write the error to eventlog. If it's a GUI app we also show the dialog.

Without this, we would have to add similar handling to what the dialog code does to figure out reasonable error message.

Technically the error code is sufficient, but I think it's better to have a nicer error even in the event log (for example it will now contain the apphost version, architecture, ...).

Reply:
I updated the comment to make this clearer.